ktor-retrofit/src/main/kotlin/com/bnorm/ktor/retrofit/RetrofitService.kt

```diff
@@ -59,6 +60,8 @@ import kotlin.reflect.full.declaredFunctions
 import kotlin.reflect.full.superclasses
 import kotlin.reflect.jvm.javaType
 
+internal val registeredRoutes = mutableMapOf<HttpMethod, Set<RegisteredRoute>>()
```

Comment:
I don't like this being a global variable. Let's install it as an `application.attribute`. Here's an example of a call attribute, but it should give you the basic idea for the pattern: https://ktor.io/samples/feature/guice.html.

Check to see if one is already installed when `retrofitService` is called and add it otherwise. This means you don't need to clear the routes between unit tests.

Reply:
Good call, cleaned up